frontend/server/workflow-helper.ts

```diff
@@ -142,18 +143,78 @@ export function toGetPodLogsStream(
   };
 }
 
+/** PartialArtifactRepositoriesValue is used to deserialize the contents of the
+ * artifact-repositories configmap.
+ */
+interface PartialArtifactRepositoriesValue {
+  s3?: {
+    keyFormat: string;
+  };
+  gcs?: {
+    keyFormat: string;
+  };
+  oss?: {
+    keyFormat: string;
+  };
+  artifactory?: {
+    keyFormat: string;
+  };
+}
+
+/**
+ * getKeyFormatFromArtifactRepositories attempts to retrieve an
+ * artifact-repositories configmap from a specified namespace. It then parses
+ * the configmap and returns a keyFormat value in its data field.
+ * @param namespace namespace of the configmap
+ */
+export async function getKeyFormatFromArtifactRepositories(
+  namespace: string,
+): Promise<string | undefined> {
+  try {
+    const [configMap] = await getConfigMap('artifact-repositories', namespace);
+    if (configMap === undefined) {
+      // If there is no artifact-repositories configmap, return undefined. The
+      // caller will just use keyFormat as specified in configs.ts.
+      return undefined;
```

Comment:
let's surface the k8serr from getConfigmap, and report that to console so it can help with troubleshooting if this part fails for whatever reasons (e.g. rbac is insufficient)

Reply:
Nice catch, @HumairAK! I pushed up a fix. Here's some manual validation evidence as well:

Workflow deleted.
<img width="220" alt="image" src="https://github.com/user-attachments/assets/fbc140ae-52d7-4348-bbf1-1cac95cd6138">

Logs still available:
<img width="1224" alt="image" src="https://github.com/user-attachments/assets/59a50417-227c-42f9-9703-774bbdf42403">

Configmap deleted:
<img width="857" alt="image" src="https://github.com/user-attachments/assets/3727acf2-8999-40ac-8092-b435fe53f4ee">

Logs not available. Error surfacing in UI pod:
<img width="914" alt="image" src="https://github.com/user-attachments/assets/e4e2f532-abba-431a-a92e-b4817b668b2a">

Is surfacing the error in the pod sufficient? We want to continue execution in case the default path specified by `ARGO_KEYFORMAT` in `configs.ts` works.

Reply:
The failing CI check seems unrelated to the code.